MetagenomeUtils.spec

```diff
@@ -12,7 +12,42 @@ module MetagenomeUtils {
         - Other functions for BinnedContig manipulation, e.g. edit_bins, merge_bins ...
     */
 
+    /* 
+      An X/Y/Z style reference
+    */
+    typedef string obj_ref;
 
+    /*
+      file_directory: file directory containing compressed/unpacked contig file(s) to build BinnedContig object
+      assembly_ref: Metagenome assembly object reference
+      binned_contig_name: BinnedContig object name
+      workspace_name: the name/id of the workspace it gets saved to
+    */
+    typedef structure {
+      string file_directory;
+      obj_ref assembly_ref;
+      string binned_contig_name;
+      string workspace_name;
+    } FileToBinnedContigParams;
 
+    typedef structure {
+      obj_ref binned_contig_obj_ref;
+    } FileToBinnedContigResult;
+
+    /*
+      file_to_binned_contigs: Generating BinnedContigs ojbect from files
+
+      input params:
+      file_directory: file directory containing compressed/unpacked contig file(s) to build BinnedContig object
```

Comment:
hmmm. I think it is generic in general. Because I suppose other apps will also generate a few files. But the way I process files in this folder is very specific to maxbin app. I think I will rename methods when we see conflicts from other apps. But for now, I think it's fine?